### PR TITLE
Improve support for AdditionalNetworkCIDRs

### PR DESCRIPTION
**Pull request by johngmyers**
Ensure that everything on AWS that tests NetworkCIDR also considers AdditionalNetworkCIDRs appropriately.

### CODE REVIEW

**Comment by johngmyers** [comment]
There's still one more issue: If one uses an API loadbalancer, one cannot put a control plane node in an AdditionalNetworkCIDR. That is because the load balancer's security group only opens the ports to the primary NetworkCIDR.

**Comment by johngmyers** [comment]
I think it might be the other way around: only nodes in the primary network CIDR can talk to apiserver or kops-controller.

File: pkg/apis/kops/validation/aws_test.go [hunk 1]
```diff
@@ -811,37 +839,45 @@ func TestAWSAdditionalRoutes(t *testing.T) {
...
-		cluster := kops.Cluster{
-			Spec: kops.ClusterSpec{
-				Networking: kops.NetworkingSpec{
-					NetworkCIDR: test.clusterCidr,
-					Subnets: []kops.ClusterSubnetSpec{
-						{
-							ID:               test.providerId,
-							Type:             test.subnetType,
-							AdditionalRoutes: test.route,
+		t.Run(test.name, func(t *testing.T) {
```
Comment (justinsb) [inline pkg/apis/kops/validation/aws_test.go:859]: Nice :+1:

File: pkg/apis/kops/validation/aws.go [hunk 2]
```diff
@@ -416,39 +404,39 @@ func hasAWSEBSCSIDriver(c kops.ClusterSpec) bool {
...
+			allErrs = append(allErrs, field.Invalid(f.Child("target"), r, "unknown target type for route"))
+		}
 
-			ipRoute, _, e := net.ParseCIDR(r.CIDR)
-			if e != nil {
-				allErrs = append(allErrs, field.Invalid(f.Child("cidr"), r, "invalid cidr"))
-			} else if clusterNet.Contains(ipRoute) && strings.HasPrefix(r.Target, "pcx-") {
-				allErrs = append(allErrs, field.Forbidden(f.Child("target"), "target is more specific than cluster CIDR block. This route can target only an interface or an instance."))
+		var routeCIDRs []*net.IPNet
+		allErrs = append(allErrs, validateCIDR(r.CIDR, f.Child("cidr"), &routeCIDRs)...)
```
Comment (justinsb) [inline pkg/apis/kops/validation/aws.go:424]: Pre-existing, but I'm not the biggest fan of the validateCIDR method signature, it's pretty hard to grok what's happening there.... might send a follow on PR to clean that up.

**Review by justinsb — APPROVED** [review]

**Comment by justinsb** [comment]
Thanks @johngmyers 

/approve
/lgtm

**Pull request merged**